Config/database/database.ts

```diff
@@ -0,0 +1,91 @@
+import RNFetchBlob from 'rn-fetch-blob';
+import { unzip } from 'react-native-zip-archive'
+import { ISchemaJSON, Schema, IConifg } from '../database/database-interfaces';
+const Realm = require('realm');
+
+let dirs = RNFetchBlob.fs.dirs
+const $dbPath = dirs.DocumentDir + '/sttmdesktop-evergreen';
+const $dbSchema = `${$dbPath}/realm-schema-evergreen.json`;
+const $md5 = `${$dbPath}/sttmdesktop.md5`;
+
+// export db variables separate from functions
+export {
+  $dbPath,
+  $dbSchema
+}
+
+// initialization functions
+
+const downloadDB = async () => {
+  console.log('execution started')
+  RNFetchBlob
+    .config({
+      fileCache: true,
+    })
+    .fetch('GET', 'https://banidb.com/databases/sttmdesktop-evergreen.zip', {
+      //some headers ..
+    })
+    // listen to download progress event
+    .progress((received, total) => {
+      console.log('progress', received / total)
```

Comment:
come up with logic to show loading bar to user...unsure how to pass to props from a currently executing func in an outside file

Reply:
have a global `downloadProgress` and `downloadFinished` variable in `database.ts` and then import it to `app.tsx` so that it updates in appState (?...._)

Reply:
that works ^